src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -683,6 +683,23 @@ protected override void ProcessRecord()
         /// </summary>
         protected override void StopProcessing() => _cancelToken?.Cancel();
 
+        /// <summary>
+        /// Cleanup disposable WebSession if it is not being kept between invocations.
+        /// </summary> 
+        protected override void EndProcessing()
+        {
+            base.EndProcessing();
+            if (!IsPersistentSession())
+            {
+                WebSession?.Dispose();
+            }
+        }
+
+        private bool IsPersistentSession()
+        {
+            return MyInvocation.BoundParameters.ContainsKey("WebSession") || !string.IsNullOrEmpty(SessionVariable);
+        }
```

Comment:
Please use the same detection for SessionVariable. Also we could use short definition with `=>`.

Reply:
Done. Have also moved into helper functions region